source/common/upstream/upstream_impl.cc

```diff
@@ -75,13 +76,26 @@ getSourceAddress(const envoy::config::cluster::v3::Cluster& cluster,
   return nullptr;
 }
 
-uint64_t parseFeatures(const envoy::config::cluster::v3::Cluster& config) {
+uint64_t
```

Comment:
Feel free to do as a follow up / TODO, but can we somehow move all of this code out of upstream_impl? It seems like somehow it should come from the protocol specific upstream code?